src/TriggerBinding/SqlTriggerListener.cs

```diff
@@ -0,0 +1,364 @@
+            {
+                await createGlobalStateTableCommand.ExecuteNonQueryAsync(cancellationToken);
+            }
+        }
+
+        /// <summary>
+        /// Inserts row for the user table and the function inside the global state table, if one does not already exist.
+        /// </summary>
+        private async Task InsertGlobalStateTableRowAsync(SqlConnection connection, SqlTransaction transaction, int userTableId, CancellationToken cancellationToken)
+        {
+            string insertRowGlobalStateTableQuery = $@"
+                IF NOT EXISTS (
+                    SELECT * FROM {SqlTriggerConstants.GlobalStateTableName}
+                    WHERE UserFunctionID = '{this._userFunctionId}' AND UserTableID = {userTableId}
+                )
+                    INSERT INTO {SqlTriggerConstants.GlobalStateTableName}
+                    VALUES ('{this._userFunctionId}', {userTableId}, CHANGE_TRACKING_MIN_VALID_VERSION({userTableId}));
+            ";
+
+            using (var insertRowGlobalStateTableCommand = new SqlCommand(insertRowGlobalStateTableQuery, connection, transaction))
+            {
+                try
+                {
+                    await insertRowGlobalStateTableCommand.ExecuteNonQueryAsync(cancellationToken);
+                }
+                catch (Exception e)
+                {
+                    // Could fail if we try to insert a NULL value into the LastSyncVersion, which happens when
+                    // CHANGE_TRACKING_MIN_VALID_VERSION returns NULL for the user table, meaning that change tracking is
+                    // not enabled for either the database or table (or both).
+
+                    string errorMessage = $"Failed to start processing changes to table {this._userTable.FullName}, " +
+                        $"potentially because change tracking was not enabled for the table or database {connection.Database}.";
+
+                    this._logger.LogWarning(errorMessage + $" Exact exception thrown is {e.Message}");
+                    throw new InvalidOperationException(errorMessage);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Creates the worker table associated with the user's table, if one does not already exist.
+        /// </summary>
+        private static async Task CreateWorkerTablesAsync(
+            SqlConnection connection,
+            SqlTransaction transaction,
+            string workerTableName,
+            IReadOnlyList<(string name, string type)> primaryKeyColumns,
+            CancellationToken cancellationToken)
+        {
+            string primaryKeysWithTypes = string.Join(", ", primaryKeyColumns.Select(col => $"{col.name} {col.type}"));
+            string primaryKeys = string.Join(", ", primaryKeyColumns.Select(col => col.name));
+
+            // TODO: Check if some of the table columns below can have 'NOT NULL' property.
+            string createWorkerTableQuery = $@"
```

Comment:
It would be good to check the primary key column names and make sure there's none that match the ones we defined as well - and if there are throw a error that indicates exactly what the problem is (otherwise that would be difficult for a user to find out)

Reply:
Did not use constants for column names since the query texts too contain literal values for column names. Having constant variables for column names inside the query strings will possibly degrade code readability.